### ISSUE TITLE
Reimplement exp2, log2 and logaddexp(2)

### ISSUE THREAD

**Issue by guilhermeleobas** [xnd-project/rbc#35]
## Description

Some methods defined by Numba are just calls to NumPy implementations. Those methods are not available in the omnisci server and may break it when one tries to use it. To circumvent this problem, we must reimplement them.

## List of Methods
- [x] `np.exp2`
- [x] `np.log2`
- [ ] `np.logaddexpr`
- [ ] `np.logaddexpr2`

## Example of a code generated

```llvm
; ModuleID = 'rbc.irtools.compile_to_IR'
source_filename = "<string>"
target triple = "x86_64-unknown-linux-gnu"

@"_ZN08NumbaEnv8__main__6fn$242Ed" = common local_unnamed_addr global i8* null

; Function Attrs: nounwind readonly
declare double @npy_log2(double) local_unnamed_addr #0

; Function Attrs: nounwind readonly
define double @fn_dadA(double %.1) local_unnamed_addr #0 {
entry:
  %.25.le.i = tail call double @npy_log2(double %.1) #1, !noalias !0
  ret double %.25.le.i
}

attributes #0 = { nounwind readonly }
attributes #1 = { nounwind }

!0 = !{!1}
!1 = distinct !{!1, !2, !"_ZN8__main__6fn$242Ed: %retptr"}
!2 = distinct !{!2, !"_ZN8__main__6fn$242Ed"}
```

**Comment by guilhermeleobas** [comment]
For `exp2` and any other function available in the C math lib, we can use the following to replace the numba implementation:
```python
from numba.targets import mathimpl # or numba.core after refactoring

mathimpl.unary_math_expr(np.exp2, "exp2", "exp2")

omni.reset()

@omni('double(double)')
def fn(x):
  return np.exp2(x)

omni.register()

_, result = omni.sql_execute('select x, fn(x) from math') # works
```

**Comment by guilhermeleobas** [comment]
First 2 methods were implemented on PR #24.

**Comment by guilhermeleobas** [comment]
Just saw this issue on Numba https://github.com/numba/numba/issues/5043

They will make it harder to redefine numba functions.
